externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/FlinkSQLOperationManager.scala

```diff
@@ -18,22 +18,38 @@
 package org.apache.kyuubi.engine.flink.operation
 
 import java.util
+import java.util.Locale
 
-import scala.collection.JavaConverters.asScalaBufferConverter
+import scala.collection.JavaConverters._
 
+import org.apache.kyuubi.config.KyuubiConf._
+import org.apache.kyuubi.config.KyuubiConf.OperationModes._
 import org.apache.kyuubi.engine.flink.result.Constants
+import org.apache.kyuubi.engine.flink.session.FlinkSessionImpl
 import org.apache.kyuubi.operation.{Operation, OperationManager}
 import org.apache.kyuubi.session.Session
 
 class FlinkSQLOperationManager extends OperationManager("FlinkSQLOperationManager") {
 
+  private lazy val operationModeDefault = getConf.get(OPERATION_PLAN_ONLY)
+
   override def newExecuteStatementOperation(
       session: Session,
       statement: String,
       confOverlay: Map[String, String],
       runAsync: Boolean,
       queryTimeout: Long): Operation = {
-    val op = new ExecuteStatement(session, statement, runAsync, queryTimeout)
+    val flinkSession = session.asInstanceOf[FlinkSessionImpl]
+    val mode =
+      flinkSession.sessionContext.getConfigMap.getOrDefault(
+        OPERATION_PLAN_ONLY.key,
+        flinkSession.normalizedConf.getOrElse(OPERATION_PLAN_ONLY.key, operationModeDefault))
```

Comment:
tuples in ` flinkSession.normalizedConf` shall be already in `flinkSession.sessionContext.getConfigMap`

Reply:
@yaooqinn, according to local debug, the `flinkSession.normalizedConf` isn't already in `flinkSession.sessionContext.getConfigMap`.

Reply:
this is a bug, session configs shall be set first in `flinkSession` for share across operations within a session.

We shall override `open` in `FlinkSessionImpl` and call it in `org.apache.kyuubi.engine.flink.session.FlinkSQLSessionManager#openSession`, which worth a separate PR to fix

Reply:
>  IMO, current implementation is correct.
>


Not exactly, 
If we don't change(merge these configs) there first,

before some `SetOperation(0)`s, the `flinkSession.normalizedConf(1)` shall update `flinkSession.sessionContext.getConfigMap(2)`, the read-side performs (1) -> (2)

After `SetOperation`s, now (2) itself contains newer settings from (0). the read-side perfroms (0) -> (1) -> (2), a.k.a a modified (2) -> (1).